pkg/apis/infinispan/v1/infinispan_types.go

```diff
@@ -58,6 +58,7 @@ type InfinispanServiceContainerSpec struct {
 	StorageClassName string `json:"storageClassName,omitempty"`
 }
 
+// +kubebuilder:validation:Enum=DataGrid;Cache
```

Comment:
@durgeshanaokar this is not enough for fix, You should add an `omitempty` annotation to the ServiceType fields.
Once you have changed all data, it's necessary to run code-generator with `make codegen`. This could change the git repo stage.

Reply:
Push the changes